<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Qualitity control and feature selection &mdash; miracle-dev 1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
        <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Configure the data" href="../tutorials/configure.html" />
    <link rel="prev" title="Preprocess" href="../tutorials/preprocess_teadog.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            miracle-dev
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../prepare.html">Preparation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data_preprocess.html">Data Preprocessing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../tutorials1.html">Tutorial: Continual Integration on Trimodality Mosaic Data (RNA+ADT+ATAC)</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../tutorials/preprocess_teadog.html">Preprocess</a><ul class="current">
<li class="toctree-l3 current"><a class="current reference internal" href="#">Qualitity control and feature selection</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/configure.html">Configure the data</a></li>
<li class="toctree-l3"><a class="reference internal" href="../tutorials/combine.html">Combine batches and generate inputs</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="teadog-4batch.html">Train with MIRACLE</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials2.html">Tutorial: Multimodal Data Label Transfer on Large-scale Reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api/MIRACLE.html">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../reproducibility.html">Reproducibility</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">miracle-dev</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../tutorials1.html">Tutorial: Continual Integration on Trimodality Mosaic Data (RNA+ADT+ATAC)</a></li>
          <li class="breadcrumb-item"><a href="../tutorials/preprocess_teadog.html">Preprocess</a></li>
      <li class="breadcrumb-item active">Qualitity control and feature selection</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/notebooks/atac+rna+adt_dogma_dig_ctrl.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container div.prompt *,
div.nboutput.container div.prompt *,
div.nbinput.container div.input_area pre,
div.nboutput.container div.output_area pre,
div.nbinput.container div.input_area .highlight,
div.nboutput.container div.output_area .highlight {
    border: none;
    padding: 0;
    margin: 0;
    box-shadow: none;
}

div.nbinput.container > div[class*=highlight],
div.nboutput.container > div[class*=highlight] {
    margin: 0;
}

div.nbinput.container div.prompt *,
div.nboutput.container div.prompt * {
    background: none;
}

div.nboutput.container div.output_area .highlight,
div.nboutput.container div.output_area pre {
    background: unset;
}

div.nboutput.container div.output_area div.highlight {
    color: unset;  /* override Pygments text color */
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars and line breaks on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
    white-space: pre;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    /*background: #f5f5f5;*/
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
    margin: 0;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt .copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
.jp-RenderedHTMLCommon table,
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
.jp-RenderedHTMLCommon thead,
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
.jp-RenderedHTMLCommon tr,
.jp-RenderedHTMLCommon th,
.jp-RenderedHTMLCommon td,
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
.jp-RenderedHTMLCommon th,
div.rendered_html th {
  font-weight: bold;
}
.jp-RenderedHTMLCommon tbody tr:nth-child(odd),
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
.jp-RenderedHTMLCommon tbody tr:hover,
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Qualitity-control-and-feature-selection">
<h1>Qualitity control and feature selection<a class="headerlink" href="#Qualitity-control-and-feature-selection" title="Permalink to this heading">ÔÉÅ</a></h1>
<p>Here we show how to preprocess a batch of data. The other three batches of data are preprocessed similarly.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">source</span><span class="p">(</span><span class="s">&quot;../preprocess/utils.R&quot;</span><span class="p">)</span>

<span class="n">base_dir</span> <span class="o">&lt;-</span> <span class="s">&quot;../data/raw/atac+rna+adt/dogma/dig_ctrl&quot;</span>
<span class="n">frag_path</span> <span class="o">&lt;-</span> <span class="nf">pj</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="s">&quot;GSM5065530_DIG_CTRL_fragments.tsv.gz&quot;</span><span class="p">)</span>
<span class="n">count_path</span> <span class="o">&lt;-</span> <span class="nf">pj</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="s">&quot;GSM5065531_DIG_CTRL_GExp_ATAC_filtered&quot;</span><span class="p">)</span>
<span class="n">adt_path</span> <span class="o">&lt;-</span> <span class="nf">pj</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="s">&quot;GSM5065532_DIG_ctrl_ADT_allCounts.mtx.gz&quot;</span><span class="p">)</span>
<span class="n">adt_bc_path</span> <span class="o">&lt;-</span> <span class="nf">pj</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="s">&quot;GSM5065532_DIG_ctrl_ADT_allCounts.barcodes.txt.gz&quot;</span><span class="p">)</span>
<span class="n">adt_feat_path</span> <span class="o">&lt;-</span> <span class="nf">pj</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="s">&quot;GSM5065532_DIG_ctrl_ADT_allCounts.proteins.txt.gz&quot;</span><span class="p">)</span>
<br/></pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Attaching SeuratObject

Registered S3 method overwritten by &#39;SeuratDisk&#39;:
  method            from
  as.sparse.H5Group Seurat

Loading required package: ensembldb

Loading required package: BiocGenerics


Attaching package: &#39;BiocGenerics&#39;


The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs


The following objects are masked from &#39;package:base&#39;:

    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which.max, which.min


Loading required package: GenomicRanges

Loading required package: stats4

Loading required package: S4Vectors


Attaching package: &#39;S4Vectors&#39;


The following object is masked from &#39;package:future&#39;:

    values


The following objects are masked from &#39;package:base&#39;:

    I, expand.grid, unname


Loading required package: IRanges

Loading required package: GenomeInfoDb

Loading required package: GenomicFeatures

Loading required package: AnnotationDbi

Loading required package: Biobase

Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&#34;Biobase&#34;)&#39;, and for packages &#39;citation(&#34;pkgname&#34;)&#39;.


Warning message:
&#34;multiple methods tables found for &#39;aperm&#39;&#34;
Warning message:
&#34;replacing previous import &#39;BiocGenerics::aperm&#39; by &#39;DelayedArray::aperm&#39; when loading &#39;SummarizedExperiment&#39;&#34;
Loading required package: AnnotationFilter


Attaching package: &#39;AnnotationFilter&#39;


The following object is masked from &#39;package:future&#39;:

    value



Attaching package: &#39;ensembldb&#39;


The following object is masked from &#39;package:stats&#39;:

    filter


Loading required package: BSgenome

Loading required package: Biostrings

Loading required package: XVector


Attaching package: &#39;Biostrings&#39;


The following object is masked from &#39;package:base&#39;:

    strsplit


Loading required package: rtracklayer

Warning message:
&#34;no function found corresponding to methods exports from &#39;BSgenome&#39; for: &#39;releaseName&#39;&#34;

Attaching package: &#39;dplyr&#39;


The following objects are masked from &#39;package:Biostrings&#39;:

    collapse, intersect, setdiff, setequal, union


The following object is masked from &#39;package:XVector&#39;:

    slice


The following objects are masked from &#39;package:ensembldb&#39;:

    filter, select


The following object is masked from &#39;package:AnnotationDbi&#39;:

    select


The following object is masked from &#39;package:Biobase&#39;:

    combine


The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union


The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect


The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union


The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union


The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union


The following objects are masked from &#39;package:stats&#39;:

    filter, lag


The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union



Attaching package: &#39;Matrix&#39;


The following object is masked from &#39;package:S4Vectors&#39;:

    expand



Attaching package: &#39;purrr&#39;


The following object is masked from &#39;package:XVector&#39;:

    compact


The following object is masked from &#39;package:GenomicRanges&#39;:

    reduce


The following object is masked from &#39;package:IRanges&#39;:

    reduce


The following object is masked from &#39;package:Signac&#39;:

    reduce


</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">output_dir</span> <span class="o">&lt;-</span> <span class="nf">pj</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="s">&quot;seurat&quot;</span><span class="p">)</span>
<span class="nf">mkdir</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span> <span class="n">remove_old</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># RNA</span>
<span class="c1"># load data</span>
<span class="n">counts</span> <span class="o">&lt;-</span> <span class="nf">Read10X</span><span class="p">(</span><span class="n">count_path</span><span class="p">)</span>
<span class="n">rna_counts</span> <span class="o">&lt;-</span> <span class="n">counts</span><span class="o">$</span><span class="n">`Gene Expression`</span>
<span class="n">atac_counts</span> <span class="o">&lt;-</span> <span class="n">counts</span><span class="o">$</span><span class="n">`Peaks`</span>
<span class="n">rna</span> <span class="o">&lt;-</span> <span class="nf">gen_rna</span><span class="p">(</span><span class="n">rna_counts</span><span class="p">)</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">rna</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_rna&quot;</span><span class="p">,</span> <span class="s">&quot;nCount_rna&quot;</span><span class="p">,</span> <span class="s">&quot;percent.mt&quot;</span><span class="p">),</span>
        <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.001</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
10X data contains more than one type and is being returned as a list containing matrices of each type.

as(&lt;dgTMatrix&gt;, &#34;dgCMatrix&#34;) is deprecated since Matrix 1.5-0; do as(., &#34;CsparseMatrix&#34;) instead

</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_4_1.png" class="no-scaled-link" src="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_4_1.png" style="width: 420px; height: 420px;" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">rna</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">rna</span><span class="p">,</span> <span class="n">subset</span> <span class="o">=</span>
    <span class="n">nFeature_rna</span> <span class="o">&gt;</span> <span class="m">300</span> <span class="o">&amp;</span> <span class="n">nFeature_rna</span> <span class="o">&lt;</span> <span class="m">4000</span> <span class="o">&amp;</span>
    <span class="n">nCount_rna</span> <span class="o">&gt;</span> <span class="m">500</span> <span class="o">&amp;</span> <span class="n">nCount_rna</span> <span class="o">&lt;</span> <span class="m">15000</span> <span class="o">&amp;</span>
    <span class="n">percent.mt</span> <span class="o">&lt;</span> <span class="m">35</span>
<span class="p">)</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">rna</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_rna&quot;</span><span class="p">,</span> <span class="s">&quot;nCount_rna&quot;</span><span class="p">,</span> <span class="s">&quot;percent.mt&quot;</span><span class="p">),</span>
        <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.001</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_5_0.png" class="no-scaled-link" src="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_5_0.png" style="width: 420px; height: 420px;" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># ADT</span>
<span class="c1"># load data</span>
<span class="n">adt_counts</span> <span class="o">&lt;-</span> <span class="nf">t</span><span class="p">(</span><span class="nf">readMM</span><span class="p">(</span><span class="n">adt_path</span><span class="p">))</span>  <span class="c1"># D * N</span>
<span class="n">adt_bc</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="n">adt_bc_path</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="n">adt_feat</span> <span class="o">&lt;-</span> <span class="nf">read.delim</span><span class="p">(</span><span class="n">adt_feat_path</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">adt_counts</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">adt_bc</span><span class="p">[,</span> <span class="m">1</span><span class="p">],</span> <span class="s">&quot;-1&quot;</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">adt_counts</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">adt_feat</span><span class="p">[,</span> <span class="m">1</span><span class="p">]</span>
<span class="n">adt_counts</span> <span class="o">&lt;-</span> <span class="n">adt_counts</span><span class="p">[,</span> <span class="nf">colnames</span><span class="p">(</span><span class="n">rna_counts</span><span class="p">)]</span>
<span class="n">adt</span> <span class="o">&lt;-</span> <span class="nf">gen_adt</span><span class="p">(</span><span class="n">adt_counts</span><span class="p">)</span>
<span class="c1"># QC</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">adt</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nCount_adt&quot;</span><span class="p">),</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.001</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_6_0.png" class="no-scaled-link" src="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_6_0.png" style="width: 420px; height: 420px;" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">adt</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">adt</span><span class="p">,</span> <span class="n">subset</span> <span class="o">=</span> <span class="n">nCount_adt</span> <span class="o">&gt;</span> <span class="m">400</span> <span class="o">&amp;</span> <span class="n">nCount_adt</span> <span class="o">&lt;</span> <span class="m">10000</span><span class="p">)</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">adt</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nCount_adt&quot;</span><span class="p">),</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.001</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> <span class="n">log</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_7_0.png" class="no-scaled-link" src="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_7_0.png" style="width: 420px; height: 420px;" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># ATAC</span>
<span class="n">atac</span> <span class="o">&lt;-</span> <span class="nf">gen_atac</span><span class="p">(</span><span class="n">frag_path</span><span class="p">)</span>
<span class="c1"># QC</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">atac</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nCount_atac&quot;</span><span class="p">,</span> <span class="s">&quot;nucleosome_signal&quot;</span><span class="p">,</span> <span class="s">&quot;TSS.enrichment&quot;</span><span class="p">),</span>
        <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.001</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">log</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Computing hash

Extracting reads overlapping genomic regions

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Fetching data...
OK

Parsing exons...
OK
Defining introns...
OK
Defining UTRs...
OK
Defining CDS...
OK

aggregating...

Done

Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message in .merge_two_Seqinfo_objects(x, y):
&#34;The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)&#34;
Warning message:
&#34;Keys should be one or more alphanumeric characters followed by an underscore, setting key from atac to atac_&#34;
Extracting TSS positions

Extracting fragments at TSSs


Computing TSS enrichment score

Rasterizing points since number of points exceeds 100,000.
To disable this behavior set `raster=FALSE`

Rasterizing points since number of points exceeds 100,000.
To disable this behavior set `raster=FALSE`

Rasterizing points since number of points exceeds 100,000.
To disable this behavior set `raster=FALSE`

Warning message:
&#34;Removed 76574 rows containing non-finite values (`stat_ydensity()`).&#34;
Warning message:
&#34;Removed 21416 rows containing missing values (`geom_point()`).&#34;
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_8_1.png" class="no-scaled-link" src="../_images/notebooks_atac+rna+adt_dogma_dig_ctrl_8_1.png" style="width: 420px; height: 420px;" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">atac</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">atac</span><span class="p">,</span> <span class="n">subset</span> <span class="o">=</span>
    <span class="n">nCount_atac</span> <span class="o">&gt;</span> <span class="m">500</span> <span class="o">&amp;</span> <span class="n">nCount_atac</span> <span class="o">&lt;</span> <span class="m">3e4</span> <span class="o">&amp;</span>
    <span class="n">nucleosome_signal</span> <span class="o">&lt;</span> <span class="m">2</span> <span class="o">&amp;</span>
    <span class="n">TSS.enrichment</span> <span class="o">&gt;</span> <span class="m">2</span>
<span class="p">)</span>
<span class="n">atac</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">atac</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">atac</span><span class="p">)[</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">atac</span><span class="o">$</span><span class="n">atac</span><span class="o">@</span><span class="n">counts</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">5</span><span class="p">])</span>
<span class="nf">VlnPlot</span><span class="p">(</span><span class="n">atac</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nCount_atac&quot;</span><span class="p">,</span> <span class="s">&quot;nucleosome_signal&quot;</span><span class="p">,</span> <span class="s">&quot;TSS.enrichment&quot;</span><span class="p">),</span>
        <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.001</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">log</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span> <span class="o">+</span> <span class="nf">NoLegend</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Get intersected cells satisfying QC metrics of all modalities</span>
<span class="n">cell_ids</span> <span class="o">&lt;-</span> <span class="nf">Reduce</span><span class="p">(</span><span class="n">intersect</span><span class="p">,</span> <span class="nf">list</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">atac</span><span class="p">),</span> <span class="nf">colnames</span><span class="p">(</span><span class="n">rna</span><span class="p">),</span> <span class="nf">colnames</span><span class="p">(</span><span class="n">adt</span><span class="p">)))</span>
<span class="n">atac</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">atac</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">cell_ids</span><span class="p">)</span>
<span class="n">rna</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">rna</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">cell_ids</span><span class="p">)</span>
<span class="n">adt</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">adt</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">cell_ids</span><span class="p">)</span>
<span class="n">atac</span>
<span class="n">rna</span>
<span class="n">adt</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># preprocess and save data</span>
<span class="nf">preprocess</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span> <span class="n">atac</span> <span class="o">=</span> <span class="n">atac</span><span class="p">,</span> <span class="n">rna</span> <span class="o">=</span> <span class="n">rna</span><span class="p">,</span> <span class="n">adt</span> <span class="o">=</span> <span class="n">adt</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../tutorials/preprocess_teadog.html" class="btn btn-neutral float-left" title="Preprocess" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../tutorials/configure.html" class="btn btn-neutral float-right" title="Configure the data" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Anonymous.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>